_posts/2020-04-30-archivesspace-cleanup-notes.md

```diff
@@ -0,0 +1,46 @@
+
+As I made progress on the basic script, Patrick suggested I incorporate string matching as a way to find and target notes for removal. Within one finding aid, notes might have small variations or extra spaces. String matching can help control how lax or restrictive you want to be when determining what notes you want to remove. Users of the edit notes script need to identify a confidence ratio, the percentage of similarity between the note string you enter into the command line and the note string that exists within the finding aid, before it can be removed. For example, if I needed to remove all versions of “open in 201X,” I would use a lower confidence ratio to ensure that all restriction notes from the 2010s were removed from a finding aid. If I wanted to ensure that only “open in 2020” notes were removed, then I could set the confidence ratio higher. I typically used a confidence ratio of 97 during my work.
+
+One of the hardest aspects of the script was incorporating argparse, a python module that allows users to provide values for variables directly in the command line. Adding argparse meant that many of my original functions within the script needed to be consolidated and rethought. Originally, the script required a lot of user input. The script would ask for one piece of information at a time. Users needed to provide Resource ID?...Note type to change?...Modify or delete notes?, etc. If I wanted to run the script over many finding aids, typing this many responses multiple times would be time consuming. With argparse, I write a single command in the command line that includes the note type (ex. access restriction note), the action choice (delete or modify), the search string (the original note in ArchivesSpace), the level of the hierarchy within a finding aid (file, item, or series), and the replacement note string if you are modifying notes. The script can then take all of this information at once, and hopefully run without issue. Essentially, argparse eliminates the need to rewrite lines of code if you want to make a minor change each time you run the script. If a command is written incorrectly, the argparse module provides help messages to get you back on track.
```

Comment:
`The script would ask for one piece of information at a time. Users needed to provide Resource ID?...Note type to change?...Modify or delete notes?, etc.`

Is it possible to combine this into a single sentence? The staccato nature of the second sentence with the short first sentence doesn't read great _to me_. If you like it though, I'd say keep it.

Reply:
sure, I can change it! Thanks for the tip!